src/rules/no-store-async.ts

```diff
@@ -0,0 +1,44 @@
+import { createRule } from "../utils"
+import type * as ESTree from "estree"
+
+export default createRule("no-store-async", {
+  meta: {
+    docs: {
+      description:
+        "disallow using async/await inside svelte stores because it causes issues with the auto-unsubscribing features",
+      category: "Possible Errors",
+      recommended: true,
+      default: "error",
+    },
+    schema: [],
+    messages: {
+      unexpected: "Do not pass async functions to svelte stores.",
+    },
+    type: "problem",
+  },
+  create(context) {
+    return {
+      CallExpression(node: ESTree.CallExpression) {
+        if (node.callee.type !== "Identifier") return
+        const { name } = node.callee
+        if (name !== "writable" && name !== "readable" && name !== "derived")
+          return
+        const [, fn] = node.arguments
+        if (fn.type !== "ArrowFunctionExpression" || !fn.async) return
```

Comment:
I think we should also check `"FunctionExpression"` ( `async function () {}` ).
(Though I don't think people will use it much.)

https://github.com/estree/estree/blob/master/es5.md#functionexpression
https://github.com/estree/estree/blob/master/es2017.md#function

Reply:
fixed!

https://github.com/ota-meshi/eslint-plugin-svelte/pull/225/commits/0b8b4ce6273f987ba0746df5c852cdf0a9b6fd5b